libmogwai-schedule/tests/connection-monitor-dummy.c

```diff
@@ -69,23 +95,52 @@ mws_connection_monitor_dummy_connection_monitor_init (MwsConnectionMonitorInterf
 static void
 mws_connection_monitor_dummy_init (MwsConnectionMonitorDummy *self)
 {
-  /* Nothing here. */
+  self->connections = g_hash_table_new_full (g_str_hash, g_str_equal, g_free,
+                                             (GDestroyNotify) connection_details_free);
+  self->cached_connection_ids = NULL;
 }
 
-static const gchar * const connection_ids[] = { NULL, };
+static void
+mws_connection_monitor_dummy_finalize (GObject *object)
+{
+  MwsConnectionMonitorDummy *self = MWS_CONNECTION_MONITOR_DUMMY (object);
+
+  g_clear_pointer (&self->connections, g_hash_table_unref);
+  g_clear_pointer (&self->cached_connection_ids, g_free);
+
+  G_OBJECT_CLASS (mws_connection_monitor_dummy_parent_class)->finalize (object);
+}
 
 static const gchar * const *
 mws_connection_monitor_dummy_get_connection_ids (MwsConnectionMonitor *monitor)
 {
-  return connection_ids;
+  MwsConnectionMonitorDummy *self = MWS_CONNECTION_MONITOR_DUMMY (monitor);
+
+  if (self->cached_connection_ids == NULL)
+    self->cached_connection_ids =
+        (const gchar **) g_hash_table_get_keys_as_array (self->connections, NULL);
+
+  return self->cached_connection_ids;
 }
 
 static gboolean
 mws_connection_monitor_dummy_get_connection_details (MwsConnectionMonitor *monitor,
                                                      const gchar          *id,
                                                      MwsConnectionDetails *out_details)
 {
-  return FALSE;
+  MwsConnectionMonitorDummy *self = MWS_CONNECTION_MONITOR_DUMMY (monitor);
+
+  const MwsConnectionDetails *found_details = g_hash_table_lookup (self->connections, id);
+
+  if (found_details != NULL)
+    {
+      connection_details_copy (found_details, out_details);
+      return TRUE;
+    }
+  else
+    {
```

Comment:
No real need for the `else` block here.

Reply:
Personal preference: keeps all the `return` statements at an equal depth.